waf_overrides.go

```diff
@@ -0,0 +1,136 @@
+package cloudflare
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// WAFOverridesResponse represents the response form the WAF overrides endpoint.
+type WAFOverridesResponse struct {
+	Response
+	Result     []WAFOverride `json:"result"`
+	ResultInfo ResultInfo    `json:"result_info"`
+}
+
+// WAFOverrideResponse represents the response form the WAF override endpoint.
+type WAFOverrideResponse struct {
+	Response
+	Result     WAFOverride `json:"result"`
+	ResultInfo ResultInfo  `json:"result_info"`
+}
+
+// WAFOverride represents a WAF override.
+type WAFOverride struct {
+	ID            string            `json:"id,omitempty"`
+	Description   string            `json:"description"`
+	URLs          []string          `json:"urls"`
+	Priority      int               `json:"priority"`
+	Groups        map[string]string `json:"groups"`
+	RewriteAction map[string]string `json:"rewrite_action"`
+	Rules         map[string]string `json:"rules"`
+	Paused        bool              `json:"paused"`
+}
+
+// ListWAFOverrides returns a slice of the WAF overrides.
+//
+// API Reference: https://api.cloudflare.com/#waf-overrides-list-uri-controlled-waf-configurations
+func (api *API) ListWAFOverrides(zoneID string) ([]WAFOverride, error) {
+	var overrides []WAFOverride
+	var res []byte
+	var err error
+
+	uri := "/zones/" + zoneID + "/firewall/waf/overrides"
+	res, err = api.makeRequest("GET", uri, nil)
+	if err != nil {
+		return []WAFOverride{}, errors.Wrap(err, errMakeRequestError)
+	}
+
+	var r WAFOverridesResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return []WAFOverride{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	if !r.Success {
+		// TODO: Provide an actual error message instead of always returning nil
```

Comment:
```suggestion
```

I think we're good to remove this. It's consistent with other methods as it is.

Reply:
Let's keep the TODO for now, because we do actually silently drop the error message here. We should wrap messages returned in `r.Errors` here.